api/providers/views.py

```diff
@@ -789,3 +793,30 @@ def get_default_queryset(self):
 
     def get_queryset(self):
         return self.get_queryset_from_request()
+
+
+class RegistrationBulkCreate(APIView):
+    parser_classes = [FileUploadParser]
+
+    permission_classes = (
+        drf_permissions.IsAuthenticated,
+        base_permissions.TokenHasScope,
+        CanAddModerator,
+    )
+
+    def put(self, request, *args, **kwargs):
+        provider_id = kwargs['provider_id']
+        file_size_limit = BULK_SETTINGS['DEFAULT_BULK_LIMIT'] * 10000
+        file_obj = request.data['file']
+        # Get hash of the file_obj
+        # Search among the existing Uploads, if an Upload with same hash exists
+        # return Response(status=409)
+        if file_obj.size > file_size_limit:
+            return Response(status=413)
```

Comment:
For the API ticket, let's use [`HTTPStatus`](https://docs.python.org/3.6/library/http.html#http.HTTPStatus).